nvdaHelper/localWin10/oneCoreSpeech.cpp

```diff
@@ -14,39 +14,38 @@ This license can be found at:
 */
 
 #include <string>
-#include <collection.h>
-#include <ppltasks.h>
-#include <wrl.h>
-#include <robuffer.h>
+#include <winrt/Windows.Media.SpeechSynthesis.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.Foundation.Metadata.h>
 #include <common/log.h>
-#include "utils.h"
 #include "oneCoreSpeech.h"
 
 using namespace std;
-using namespace Platform;
-using namespace Windows::Media::SpeechSynthesis;
-using namespace concurrency;
-using namespace Windows::Storage::Streams;
-using namespace Microsoft::WRL;
-using namespace Windows::Media;
-using namespace Windows::Foundation::Collections;
-using Windows::Foundation::Metadata::ApiInformation;
+using namespace winrt;
+using namespace winrt::Windows::Media::SpeechSynthesis;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Media;
+using namespace winrt::Windows::Foundation::Collections;
+using winrt::Windows::Foundation::Metadata::ApiInformation;
 
 bool __stdcall ocSpeech_supportsProsodyOptions() {
-	return ApiInformation::IsApiContractPresent("Windows.Foundation.UniversalApiContract", 5, 0);
+	return ApiInformation::IsApiContractPresent(hstring{L"Windows.Foundation.UniversalApiContract"}, 5, 0);
 }
 
-OcSpeech* __stdcall ocSpeech_initialize() {
-	auto instance = new OcSpeech;
-	instance->synth = ref new SpeechSynthesizer();
+OcSpeech::OcSpeech() : synth(SpeechSynthesizer{}) {
 	// By default, OneCore speech appends a  large annoying chunk of silence at the end of every utterance.
 	// Newer versions of OneCore speech allow disabling this feature, so turn it off where possible.
-	if (ApiInformation::IsApiContractPresent("Windows.Foundation.UniversalApiContract", 6, 0)) {
-		auto options = instance->synth->Options;
-		options->AppendedSilence = SpeechAppendedSilence::Min;
+	if (ApiInformation::IsApiContractPresent(hstring{L"Windows.Foundation.UniversalApiContract"}, 6, 0)) {
+		synth.Options().AppendedSilence(SpeechAppendedSilence::Min);
 	} else {
 		LOG_DEBUGWARNING(L"AppendedSilence not supported");
 	}
+}
+
+OcSpeech* __stdcall ocSpeech_initialize() {
+	auto instance = new OcSpeech;

```

Comment:
As per OCR, let's reconsider this approach.

Reply:
We can follow up on this via new issue: https://github.com/nvaccess/nvda/issues/13253